### PR TITLE
Slash Commands

### PR DESCRIPTION
**Pull request by JBraunsmaJr**
Entire framework was modified

Database and configuration files now live within a data folder. Makes it easier to mount and move to other hosts
All commands now use the slash command interaction system

Supported commands

/join
/leave
/invite
/archive (#19 #18  -- will complete intended functionality)
/create-reminder
/delete-reminder
/show-reminders
/create-class

### CODE REVIEW

File: DevryApplication/ServiceCollectionExtensions.cs [hunk 1]
```diff
@@ -0,0 +1,14 @@
+﻿using DevryApplication.Tasks.Scheduling;
```
Comment (JBraunsmaJr) [inline DevryApplication/ServiceCollectionExtensions.cs:1]: Delete file

File: DevryApplication/Tasks/Scheduling/SchedulerTaskExtensions.cs [hunk 2]
```diff
@@ -0,0 +1,26 @@
...
+using System.Threading.Tasks;
+using Microsoft.Extensions.DependencyInjection;
+using Microsoft.Extensions.Hosting;
+
+namespace DevryApplication.Tasks.Scheduling
+{
+    public static class SchedulerTaskExtensions
+    {
+        public static IServiceCollection AddScheduler(this IServiceCollection services)
+            => services.AddSingleton<SchedulerBackgroundService>();
```
Comment (JBraunsmaJr) [inline DevryApplication/Tasks/Scheduling/SchedulerTaskExtensions.cs:11]: Should change this to be AddHostedService

File: DevryBot/ConfigurationExtensions.cs [hunk 3]
```diff
@@ -0,0 +1,87 @@
...
+        }
+
+        /// <summary>
+        /// Retrieve list of majors that are available to the 
+        /// </summary>
+        /// <param name="config"></param>
+        /// <returns></returns>
+        public static List<string> Majors(this IConfiguration config)
+        {
+            return config.GetEnumerable("Discord:Majors").ToList();
```
Comment (JBraunsmaJr) [inline DevryBot/ConfigurationExtensions.cs:35]: Add slash command for picking major
Add major list into configuration file

File: DevryBot/Discord/Interactions/MajorCommand.cs [hunk 4]
```diff
@@ -0,0 +1,42 @@
...
+using System.Threading.Tasks;
+using DSharpPlusNextGen;
+using DSharpPlusNextGen.Entities;
+using DSharpPlusNextGen.Enums;
+using DSharpPlusNextGen.EventArgs;
+using DiscordMember = DSharpPlus.Entities.DiscordMember;
+
+namespace DevryBot.Discord.Interactions
+{
+    public static class MajorCommand
```
Comment (JBraunsmaJr) [inline DevryBot/Discord/Interactions/MajorCommand.cs:12]: Remove this from this MR
-- Out of scope

**Review by JBraunsmaJr — COMMENTED** [review]
Apply changes

File: DevryBot/ConfigurationExtensions.cs (hunk 3 above)
Comment (JBraunsmaJr) [reply DevryBot/ConfigurationExtensions.cs:35]: #21 - shall implement this feature

**Comment by tyates2020** [comment]
Approve

**Review by tyates2020 — APPROVED** [review]

**Pull request merged**